tests/unit/test_util.py

```diff
@@ -50,3 +51,52 @@ def test_calculate_time_delta(input, expect):
         assert res is None
     else:
         assert res == expect
+
+
+@pytest.mark.parametrize(
+    "input_roles,role,expected_result",
+    [
+        ([], enums.Role.VERIFIED, False),
+        ([enums.Role.MOD], enums.Role.VERIFIED, False),
+        ([enums.Role.VERIFIED], enums.Role.VERIFIED, True),
+        ([enums.Role.MOD, enums.Role.VERIFIED], enums.Role.VERIFIED, True),
+    ],
+)
+def test_user_has_role(
+    input_roles: List[str],
+    role: enums.Role,
+    expected_result: bool,
+    mocker: MockerFixture,
+    create_role,
+):
+    roles = [create_role(name=role_name) for role_name in input_roles]
+    mocked_member = mocker.Mock(roles=roles)
+
+    res = util.user_has_role(mocked_member, role)
+
+    assert res == expected_result
+
+
+async def test_add_and_remove_role(mocker: MockerFixture, naur_guild):
+    role_to_add = enums.Role.EXILED
+    role_to_remove = enums.Role.VERIFIED
+
+    mock_add_roles = mocker.AsyncMock()
+    mock_remove_roles = mocker.AsyncMock()
+
+    # NB eventually making a mocked Member should be a fixture, but I want to understand
+    # needs a bit more before doing so
```

Comment:
im sorry these two lines are giving me a stroke

Reply:
Sorry, I just wanted to call out that `Member` should be something that we can create easy mocks for. Probably we should do this via a factory method wrapped in a fixture, like I have set up for `Role` currently. I didn't put that together quite yet because I wanted to see more use cases before fully refactoring